src/js/components/visualization/ChartMenu.jsx

```diff
@@ -0,0 +1,35 @@
+import changeCase from 'change-case'
+import Download from 'material-ui/svg-icons/file/file-download'
+import MenuItem from 'material-ui/MenuItem/MenuItem'
+import {saveCanvas} from '../../modules/utilities'
+import React, {Component, PropTypes} from 'react'
+
+export class ChartMenu extends Component {
+  static propTypes = {
+    metadata: PropTypes.object.isRequired,
+    visualization: PropTypes.object.isRequired
+  }
+
+  render () {
+    const {metadata, visualization} = this.props
+    
+    return (
+      <span>
+        <MenuItem
+          key='saveChart'
+          leftIcon={<Download />}
+          primaryText='Save'
+          onTouchTap={function (event) {
```

Comment:
don't define callbacks inline. when you do, they get created on every render call. if we find this elsewhere in the code, it should be re-worked if possible